examples/long-short/long-short.go

```diff
@@ -108,57 +131,69 @@ func (alp alpacaClientContainer) run() {
 			}
 			qty, _ := position.Qty.Float64()
 			qty = math.Abs(qty)
-			alpacaClient.submitOrder(int(qty), position.Symbol, orderSide)
+			if err := algo.submitOrder(int(qty), position.Symbol, orderSide); err != nil {
+				return fmt.Errorf("submit order: %w", err)
+			}
 		}
 		// Run script again after market close for next trading day.
-		fmt.Println("Sleeping until market close (15 minutes).")
+		fmt.Println("Sleeping until market close (15 minutes)")
 		time.Sleep(15 * time.Minute)
 	} else {
 		// Rebalance the portfolio.
-		alpacaClient.rebalance()
+		if err := algo.rebalance(); err != nil {
+			fmt.Println("Failed to rebalance, will try again in a minute:", err)
+		}
+		fmt.Println("Sleeping for 1 minute")
 		time.Sleep(1 * time.Minute)
 	}
+	return nil
 }
 
 // Spin until the market is open.
-func (alp alpacaClientContainer) awaitMarketOpen() bool {
-	clock, _ := alpacaClient.client.GetClock()
+func (alp longShortAlgo) awaitMarketOpen() (bool, error) {
```

Comment:
I think this could sleep until the market is open.

Reply:
I was trying to change as little as possible (sounds funny, I know). I think this is okey as it is.